scripts/check-and-run-apollo-cli.sh

```diff
@@ -17,18 +17,33 @@ if [[ -z "$CONFIGURATION" ]]; then
   exit 1
 fi
 
-# Define NVM_DIR and source the nvm.sh setup script
-[[ -z "$NVM_DIR" ]] && export NVM_DIR="$HOME/.nvm"
+use_correct_node_via_nvm() {
+  nvm version $REQUIRED_NODE_VERSION > /dev/null
+  if [[ $? -eq 0 ]]; then
+    # The version of node that we want is installed.
+    nvm use $REQUIRED_NODE_VERSION
+  else
+    nvm install $REQUIRED_NODE_VERSION
+  fi
+}
+
+use_correct_node_via_nodenv() {
+  nodenv install -s $REQUIRED_NODE_VERSION
+  nodenv shell $REQUIRED_NODE_VERSION
+}
 
 if [[ -s "$HOME/.nvm/nvm.sh" ]]; then
   . "$HOME/.nvm/nvm.sh"
+  use_correct_node_via_nvm
 elif [[ -x "$(command -v brew)" && -s "$(brew --prefix nvm)/nvm.sh" ]]; then
   . "$(brew --prefix nvm)/nvm.sh"
+  use_correct_node_via_nvm
 fi
 
 # Set up the nodenv node version manager if present
 if [[ -x "$HOME/.nodenv/bin/nodenv" ]]; then
```

Comment:
Full transparency: I just installed a clean version of Nodenv and this file doesn't exist, so I am wondering if this check is incorrect. That being said, I've tested the commands here for installing Node 8.15.0 via Nodenv and they work. Determining if in fact there is an issue in this check should be a separate investigation / PR.